src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1246,5 +1251,59 @@ private void VerifyIsNotZombie()
         {
             ErrorUtilities.VerifyThrow(!_isZombie, "RequestBuilder has been zombied.");
         }
+
+        /// <summary>
+        /// Configure warnings as messages and errors based on properties defined in the project.
+        /// </summary>
+        private void ConfigureWarningsAsErrorsAndMessages()
+        {
+            // Gather needed objects
+            //
+            ProjectInstance project = _requestEntry?.RequestConfiguration?.Project;
+            BuildEventContext buildEventContext = _projectLoggingContext?.BuildEventContext;
+            ILoggingService loggingService = _projectLoggingContext?.LoggingService;
+
+            // Ensure everything that is required is available at this time
+            //
+            if (project != null && buildEventContext != null && loggingService != null && buildEventContext.ProjectInstanceId != BuildEventContext.InvalidProjectInstanceId)
+            {
+                if (String.Equals(project.GetPropertyValue(MSBuildConstants.TreatWarningsAsErrors)?.Trim(), "true", StringComparison.OrdinalIgnoreCase))
+                {
+                    // If <MSBuildTreatWarningsAsErrors was specified then an empty ISet<string> signals the IEventSourceSink to treat all warnings as errors
+                    //
+                    loggingService.AddWarningsAsErrors(buildEventContext.ProjectInstanceId, new HashSet<string>());
+                }
+                else
+                {
+                    ISet<string> warningsAsErrors = ParseWarningCodes(project.GetPropertyValue(MSBuildConstants.WarningsAsErrors));
+
+                    if (warningsAsErrors?.Count > 0)
+                    {
+                        loggingService.AddWarningsAsErrors(buildEventContext.ProjectInstanceId, warningsAsErrors);
+                    }
+                }
+
+                ISet<string> warningsAsMessages = ParseWarningCodes(project.GetPropertyValue(MSBuildConstants.WarningsAsMessages));
```

Comment:
Suppressing a warning overrides elevating one.  This is so you can treat all warnings as errors but still turn a few of them off.